app/controllers/customers_controller.rb

```diff
@@ -85,12 +90,17 @@ def update
         :logged_in_id => current_user.id,
         :comments => flash[:notice])
       if @customer.email_changed? && @customer.valid_email_address? &&
-          params[:dont_send_email].blank? 
+          params[:dont_send_email].blank?
```

Comment:
I am removing this line completely. There's already an after-save action that does this so changing the customer email automatically changes the identity uid. When the params[:dont_sen_email] was blank, it would just update twice which has the same effect as updating once, so I didn't notice it.